### ISSUE TITLE
[📖] ForwardRef alternatives in Qwik

### ISSUE THREAD

**Issue by Aetherovox** [BuilderIO/qwik#3046]
### Suggestion

I was looking at porting react-spring over to Qwik as my first (although ambitious) library publication. 
React has the ForwardRef type in order to pass a ref to a child component. 
It would be helpful to have a section for React users on how to build out libraries in quick, with a cheat-sheet for alternative types that one could use. 
This should accelerate the development of the wider ecosystem.

**Comment by zanettin** [comment]
@manucorporat not sure if qwik offers smth in this direction.
https://reactjs.org/docs/forwarding-refs.html

**Comment by nnelgxorz** [comment]
I'm probably missing something. Couldn't the child component just take a signal as part of props?

**Comment by gioboa** [comment]
Hi @Aetherovox, I implemented the solution with Qwik 1.2.10. btw a lot of things happened during this months. 🚀
Basically you can use props.
```
import { type Signal, component$, useSignal, useVisibleTask$ } from '@builder.io/qwik';

export default component$(() => {
 const myButtonRefSig = useSignal<Element>();
 useVisibleTask$(() => {
  if (myButtonRefSig) {
   myButtonRefSig.value?.setAttribute('style', 'border: 2px red solid');
  }
 });
 return (
  <div>
   <MyButton forwardRef={myButtonRefSig} />
  </div>
 );
});

export const MyButton = component$<{ forwardRef: Signal<Element | undefined> }>(({ forwardRef }) => {
  return <button ref={forwardRef}>My button</button>;
});
```
So I'm closing this issue for now, feel free to re-open it if it's still an issue for you.
